hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/router/TestRouterRpc.java

```diff
@@ -2070,6 +2077,46 @@ public void testSetBalancerBandwidth() throws Exception {
     }, 100, 60 * 1000);
   }
 
+  @Test
+  public void testRecordRealClient() throws Exception {
```

Comment:
Does it make sense to refactor this into a general function and call it with the two configs?
The other interesting part would be to log the output too.

Reply:
Thanks @goiri  for the comment and review.
If you want HADOOP_CALLER_CONTEXT_ENABLED_KEY to reset and take effect, MiniRouterDFSCluster needs to be restarted, so I refactored the process of starting MiniRouterDFSCluster. And use a static Configuration, which is for the NameNode.
You mean keep globalSetUp() unchanged and add static Configuration?
Another question is, do you think the new format of logging needs to be updated?